src/dashboard.tsx

```diff
@@ -116,7 +116,25 @@ export class DaskDashboardLauncher extends Widget {
     this.addClass('dask-DaskDashboardLauncher');
     this._items = options.items || DaskDashboardLauncher.DEFAULT_ITEMS;
     this._launchItem = options.launchItem;
-    this._input.urlChanged.connect(this.update, this);
+    this._input.urlChanged.connect(this.updateLinks, this);
+  }
+
+  private async updateLinks(): Promise<void> {
+    const result = await Private.getItems(
+      this._input.url,
+      ServerConnection.makeSettings()
```

Comment:
What do you think about making the settings once, keeping them as a private member, and then passing them into the `URLInput` as a constructor arg? I don't think that it's particularly expensive, nor is this function called all that often, but creating them de novo each time rubs be a bit the wrong way.

Reply:
This is your project and I defer to your architectural preferences entirely!

Honestly, from a design standpoint, my preference to not keep a reference to `ServerConnection.makeSettings()` anywhere, but call it any time it is needed. `private _serverSettings: ServerConnection.ISettings` implies that `_serverSettings` is a necessary state variable to `URLInput`, something it will use either as permutable state or as implicit "environment state" (the banana is held by the gorilla). But the settings belong to `ServerConnection`, and should be left to it to coordinate.

Reply:
I made that change as you suggested. Doesn't it seem messy, compared to leaving the `makeSettings()` as the responsibility of `ServerConnection`? Now we've got two different classes with their own private `_serverSettings` objects and potential risk of state bugs relating to them falling out of sync. In an OO context it implies (Please lets not do this!) that there should be a parent object `ServerSettingsStateHolder` that `URLInput` and `DaskDashboardLauncher` both inherit from so that we don't have to dupe all the `_serverSettings` member code.

Reply:
The `ISettings` interface is entirely marked as `readonly`, so at least we should be able to be confident that it won't mutate out from under us: https://github.com/jupyterlab/jupyterlab/blob/1f59d9c1b1e1cae9323e3fa9b3b3a4a98e20ca84/packages/services/src/serverconnection.ts#L53-L101 . I think I also disagree that they belong to `serverConnection`, since they are passed in by the caller of `makeRequest`. I think of `makeSettings` as more of a convenience method for creating default values.

We can make `serverSettings` an optional input for `URLInput`, and it can create its own version if it is not provided. My main goal here is to ensure that the dashboard and the `URLInput` are using the same settings for talking to the server. Right now, we are using the defaults, so they wind up being identical. But if in the future we want to allow that to be customizable, it makes it somewhat easier for them to share a reference to the same settings.

It's not really a big deal either way, though.

Reply:
I've made the changes that allow them to share the result of `DaskDashboardLauncher` requesting the `makeSettings` object. I'm fine to submit it that way if you like it!